src/main/java/com/redislabs/redisai/Model.java

```diff
@@ -0,0 +1,215 @@
+            inputs = new String[inputsEncoded.size()];
+            for (int j = 0; j < inputsEncoded.size(); j++) {
+              inputs[j] = SafeEncoder.encode(inputsEncoded.get(j));
+            }
+          }
+          break;
+        case "outputs":
+          List<byte[]> outputsEncoded = (List<byte[]>) reply.get(i + 1);
+          if (outputsEncoded.size() > 0) {
+            outputs = new String[outputsEncoded.size()];
+            for (int j = 0; j < outputsEncoded.size(); j++) {
+              outputs[j] = SafeEncoder.encode(outputsEncoded.get(j));
+            }
+          }
+          break;
+        default:
+          break;
+      }
+    }
+    if (backend != null && device != null && blob != null) {
+      model = new Model(backend, device, inputs, outputs, blob);
+      if (tag != null) {
+        model.setTag(tag);
+      }
+      if (batchsize > -1) {
+        model.setBatchSize(batchsize);
+      }
+      if (minbatchsize > -1) {
+        model.setMinBatchSize(minbatchsize);
+      }
+    }
+    return model;
+  }
+
+  public String getTag() {
+    return tag;
+  }
+
+  public void setTag(String tag) {
+    this.tag = tag;
+  }
+
+  public byte[] getBlob() {
+    return blob;
+  }
+
+  public void setBlob(byte[] blob) {
+    this.blob = blob;
+  }
+
+  public String[] getOutputs() {
+    return outputs;
+  }
+
+  public void setOutputs(String[] outputs) {
+    this.outputs = outputs;
+  }
+
+  public String[] getInputs() {
+    return inputs;
+  }
+
+  public void setInputs(String[] inputs) {
+    this.inputs = inputs;
+  }
+
+  public Device getDevice() {
+    return device;
+  }
+
+  public void setDevice(Device device) {
+    this.device = device;
+  }
+
+  public Backend getBackend() {
+    return backend;
+  }
+
+  public void setBackend(Backend backend) {
+    this.backend = backend;
+  }
+
+  public long getBatchSize() {
+    return batchSize;
+  }
+
+  public void setBatchSize(long batchsize) {
+    this.batchSize = batchsize;
+  }
+
+  public long getMinBatchSize() {
+    return minBatchSize;
+  }
+
+  public void setMinBatchSize(long minbatchsize) {
+    this.minBatchSize = minbatchsize;
+  }
+
+  /**
+   * Encodes the current model properties into an AI.MODELSET command to be store in RedisAI Server
+   *
+   * @param key name of key to store the Model
+   * @return
+   */
+  protected List<byte[]> getModelSetCommandBytes(String key) {
+    List<byte[]> args = new ArrayList<>();
+    args.add(SafeEncoder.encode(key));
+    args.add(backend.getRaw());
+    args.add(device.getRaw());
+    if (tag != null) {
+      args.add(Keyword.TAG.getRaw());
+      args.add(SafeEncoder.encode(tag));
+    }
+    if (batchSize > 0) {
+      args.add(Keyword.BATCHSIZE.getRaw());
+      args.add(Protocol.toByteArray(batchSize));
+      if (minBatchSize > 0) {
+        args.add(Keyword.MINBATCHSIZE.getRaw());
+        args.add(Protocol.toByteArray(minBatchSize));
+      }
+    }
+    args.add(Keyword.INPUTS.getRaw());
```

Comment:
Aren't INPUTS/OUTPUTS optional?  I thought it's only relevant for TF

Reply:
`INPUTS` and `OUTPUTS` are mandatory ( even if empty for all except TF )